web/app/components/user-info/user-info.jsx

```diff
@@ -5,6 +5,7 @@ import api from 'common/api';
 import { getHandleClickProps } from 'common/accessibility';
 
 import LastCommentsList from './last-comments-list';
+const CommentAvatarSrc = require('../comment/__avatar/comment__avatar.svg');
```

Comment:
cc @igoradamenko need your opinion here. How to share svg between components without importing from others components element.

Reply:
I usually create a BEM block svg-icon, which I use for all svg-icons in a projects. 

It looks like this:

```jsx
// svg-icon.jsx
export default class SVGIcon extends PureComponent {
  render() {
    const { symbol } = this.props;

    if (!symbol) return null;

    return (
      <svg className={b('svg-icon', this.props)} focusable="false">
        <use xlinkHref={`#${symbol.default.id}`}/>
      </svg>
    );
  }
}

SVGIcon.propTypes = {
  symbol: PropTypes.object,
};
```

```scss
// svg-icon.scss
.svg-icon {
  max-width: 100%;
  max-height: 100%;
  stroke: currentColor;
  fill: currentColor;
}
```

And I keep all symbols as elements on this block:

```svg
<!-- svg-icon/__symbol/_type/_phone/svg-icon__symbol_type_phone.svg -->
<svg xmlns="http://www.w3.org/2000/svg" width="25" height="31" viewBox="0 0 25 31">
  <path stroke="none" fill-rule="evenodd" d="M23.884 25.842c.195.195.347.42.454.674.108.254.16.527.16.82 0 .332-.072.644-.218.937-.147.293-.347.538-.6.733-.294.273-.616.513-.968.718-.35.205-.713.366-1.084.483-.47.117-.937.2-1.406.25-.47.048-.947.072-1.436.072h-.03c-1.952 0-3.76-.39-5.42-1.18-1.66-.78-3.124-1.68-4.393-2.7-1.27-1.02-2.32-2.01-3.15-2.99-.83-.98-1.39-1.67-1.684-2.08-1.49-2.07-2.48-4.15-2.98-6.23s-.71-4-.62-5.79c.08-1.79.38-3.35.88-4.69.49-1.34 1-2.31 1.5-2.92.27-.26.56-.49.86-.69.31-.21.63-.39.98-.54.14-.08.29-.14.44-.18.16-.04.33-.06.5-.06.12 0 .24.01.35.03.12.02.23.04.33.08.27.1.51.24.73.44.21.19.38.42.5.67l2.23 5.01c.08.13.13.28.16.42.03.14.04.3.04.48 0 .45-.12.85-.37 1.2s-.55.62-.93.82l-.68.26c-.3.13-.54.34-.74.61-.198.27-.3.57-.32.91 0 .1-.01.2-.02.31-.01.11-.016.22-.016.34 0 1.01.142 1.98.44 2.9.29.91.69 1.75 1.2 2.52l.79 1.31c.7.95 1.54 1.77 2.502 2.46.97.68 2.034 1.2 3.21 1.55.06.02.11.03.16.04.05.01.102.01.16.01.173 0 .33-.03.47-.09.133-.06.26-.147.38-.26l.85-.79c.173-.18.38-.317.63-.41.24-.1.5-.15.773-.15.29 0 .57.05.833.16s.48.26.66.45l3.84 4.02zM22.42 27.57h.028c.04-.02.074-.05.103-.088.03-.04.05-.087.05-.146 0-.04 0-.073-.01-.103s-.03-.063-.07-.102l-3.81-3.98c-.02-.02-.04-.03-.05-.04-.02-.01-.05-.01-.08-.01-.02 0-.03.01-.04.02l-.04.05-.85.79c-.29.28-.62.49-.98.63-.36.15-.75.22-1.16.22-.15 0-.31-.01-.47-.03-.15-.02-.32-.06-.5-.12-1.35-.41-2.59-1.02-3.73-1.84-1.15-.82-2.13-1.8-2.95-2.93l-.79-1.29c-.62-.93-1.1-1.95-1.43-3.04-.33-1.09-.5-2.23-.5-3.42v-.76c.06-.7.29-1.32.69-1.87.4-.54.91-.96 1.51-1.26l.65-.29c.06-.02.1-.05.13-.1s.047-.1.047-.16v-.1c0-.03-.01-.05-.03-.07l-2.25-5-.054-.06c-.02-.02-.04-.03-.06-.03 0-.02-.004-.03-.013-.03H5.7c-.02 0-.038.01-.058.02-.02.01-.04.016-.06.016-.234.097-.454.214-.66.35-.205.138-.395.284-.57.44-.32.373-.69 1.113-1.12 2.227C2.8 6.57 2.537 7.917 2.44 9.5c-.1 1.58.067 3.324.497 5.23.43 1.903 1.338 3.832 2.724 5.785 1.67 2.325 3.34 4.058 5.03 5.2 1.69 1.144 3.25 1.925 4.68 2.345 1.43.42 2.66.6 3.69.542 1.04-.06 1.73-.127 2.08-.205.26-.097.49-.214.71-.35.22-.138.41-.294.59-.47z"/>
</svg>
```

When I need to inject svg-icon it looks like this:

```jsx
<SVGIcon
  symbol={require('app/components/svg-icon/__symbol/_type/_phone/svg-icon__symbol_type_phone.svg')}
  mix="some-block__some-element"
/>
```

So, to make it work I usually add svg-sprite-loader to a webpack config:

```js
{
  test: /svg-icon__symbol.+\.svg$/,
  loader: 'svg-sprite-loader',
  options: {
    symbolId: 'svg-symbol-[name]-[hash]',
  },
},
```

(It matches svg-icon's symbols only, so I can inject any other svgs using regular file-loader with exclude option, if I need it.)

So, this way have some pros and cons.

Pros:
1. Splitting. If you split a bundle, webpack splits svgs between parts of this bundle without any problem.
2. Reuse. You can easy reuse svg-icon's without duplicating icons in a bundle.

Cons:
1. You need to use explicit path to the svg icon without any variables. If you don't and require using the way like this: `require('svg-icon/__symbol/_type/_' + iconName + '/svg-icon__symbol_type_' + iconName + '.svg')`, webpack will add all icons of svg-icon/__symbol directory into a bundle. 

It works for me, because it looks like parameterization of a block, but in some specific way, because of webpack, splitting, bundling, etc.

Reply:
It's up to you. If you think that there won't be a lot of icons in the future, yes, maybe it's better to create a folder like `assets/icon` and keep it there. Also it's possible to create an independent block `avatar-icon` which will require this icon and export it.

Reply:
Also in this case I think it's better to separate avatar entity to a new block and use it everywhere. And this `avatar` will require svg placeholder, etc.

Reply:
@lexich Lets extract Avatar as independent component

Reply:
@Guria Ok